engine/src/test/java/engine/physic/PhysicManagerTest.java

```diff
@@ -30,11 +31,27 @@ public void shouldThrowWhenInsertedObjectOutOfBoundary() throws AddObjectExcepti
     }
 
     @Test(expected = AddObjectException.class)
-    public void shouldThrowWhenInsertedObjectIntersectsOthers() throws AddObjectException {
-        physicManager.addPhysicObject(new GameObject(new Vector(1, 1), new Vector(3, 3), 0, 0));
-        GameObject gameObject = new GameObject(new Vector(2, 2), new Vector(4, 4), 0, 0);
-
+    public void shouldThrowWhenInsertedSolidObjectIntersectsOthersSolidObjects() throws AddObjectException {
+        GameObject gameObject = new GameObject(new Vector(1, 1), new Vector(3, 3), 0, 0);
+        gameObject.setSolid(true);
         physicManager.addPhysicObject(gameObject);
+        GameObject otherObject = new GameObject(new Vector(2, 2), new Vector(4, 4), 0, 0);
+        otherObject.setSolid(true);
+
+        physicManager.addPhysicObject(otherObject);
+    }
+
+    @Test
+    public void shouldNotThrowWhenInsertedObjectIsNotSolid() {
+        GameObject gameObject = new GameObject(new Vector(1, 1), new Vector(2, 2), 0, 0);
+        gameObject.setSolid(false);
+
+        try {
+            physicManager.addPhysicObject(gameObject);
+
+        } catch (AddObjectException e) {
+            fail();
```

Comment:
test is going to fail itself when exception is thrown. No need to catch it and call other methods